### PR TITLE
feat: add helpers to run banner auctions

### PR DESCRIPTION
**Pull request by fcs-ts**
This should allow avoiding the creation of Auction objects manually, which gets a lot harder than for listings (although maybe we can do similar helpers for listings).

TODO:
- [x] tests
- [x] figure out why `app` starts to break with this changes (It seems they break even for `main` in my setup)

### CODE REVIEW

File: TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerConfig.kt [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+     *
+     * @property slotId id of the banner slot
+     * @property ids ids of the entities that are competing for the banner
+     * @property device can be "desktop" or "mobile"
+     * @property geoTargeting optional location for geo-targeted banners
+     */
+    data class LandingPage(
+        val slotId: String,
+        val ids: List<String>,
+        val device: String? = null,
```
Comment (fcs-ts) [inline TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerConfig.kt]: I think we should use an enum for device, what do you think @anonvt?

Comment (PabloReszczynski) [reply TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerConfig.kt]: I'd add that device should be set to `mobile` by default, as in most cases that will be the intended use case.

Comment (fcs-ts) [reply TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerConfig.kt]: you are right, I eas matching the optional from the api (which in turn is converted to desktop internally), but definitely, mobile should be the default here

Comment (anonvt) [reply TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerConfig.kt]: Agreed that an enum would be the best fit.

**Review by anonvt — APPROVED** [review]
I pulled and rebased off of main, :app runs tests (so not broken?)

**Comment by anonvt** [comment]
This includes #15 correct?

**Comment by fcs-ts** [comment]
> This includes https://github.com/Topsort/topsort.kt/pull/15 correct?

It's the other way around, it's the part of #15 that doesn't include the custom View

File: TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerConfig.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+     * @property slotId id of the banner slot
+     * @property ids ids of the entities that are competing for the banner
+     * @property device target device for the banner
+     * @property geoTargeting optional location for geo-targeted banners
+     */
+    data class LandingPage(
+        val slotId: String,
+        val ids: List<String>,
+        val device: Device = Device.MOBILE,
+        val geoTargeting: String? = null
```
Comment (PabloReszczynski) [inline TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerConfig.kt:22]: Does this convert to `geoTargeting: { location: string }` in the final payload?

Comment (fcs-ts) [reply TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerConfig.kt:22]: yes, the `build*` functions in the `Auction` class do that for you :)

**Review by PabloReszczynski — APPROVED** [review]

**Pull request merged**